plugin.rb

```diff
@@ -509,10 +513,12 @@
   add_to_serializer(:post, :activity_pub_enabled) do
     object.activity_pub_enabled
   end
-  activity_pub_post_custom_fields.each do |field_name|
-    add_to_serializer(:post, "activity_pub_#{field_name}".to_sym) do
-      object.send("activity_pub_#{field_name}")
-    end
+  activity_pub_post_custom_field_names.each do |field_name|
+    add_to_serializer(
+      :post,
+      field_name.to_sym,
+      include_condition: -> { object.activity_pub_enabled }
```

Comment:
This is nice. IIUC, this cleans up the JSON output of topics where AP isn't enabled.

Reply:
Yup, that's right, it also guards against unnecessary work (queries) when the plugin is disabled.